tests/checkpoint/test_checkpoint_with_fluent_datasources.py

```diff
@@ -1037,3 +1041,1495 @@ def test_checkpoint_run_adds_validation_ids_to_expectation_suite_validation_resu
+        **{
+            "datasource_name": "my_spark_dataframes_datasource",
+            "data_asset_name": "my_dataframe_asset",
+        }
+    )
+    checkpoint: Checkpoint = Checkpoint(
+        name="my_checkpoint",
+        data_context=context,
+        config_version=1,
+        run_name_template="%Y-%M-foo-bar-template",
+        expectation_suite_name="my_expectation_suite",
+        action_list=common_action_list,
+    )
+    with pytest.raises(
+        gx_exceptions.DataContextError, match=r"expectation_suite .* not found"
+    ):
+        checkpoint.run(validations=[{"batch_request": batch_request}])
+
+    assert len(context.validations_store.list_keys()) == 0
+
+    context.add_expectation_suite("my_expectation_suite")
+    result = checkpoint.run(validations=[{"batch_request": batch_request}])
+
+    assert len(context.validations_store.list_keys()) == 1
+    assert result["success"]
+
+
+@pytest.mark.integration
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+@pytest.mark.slow  # 1.31s
+def test_newstyle_checkpoint_instantiates_and_produces_a_validation_result_when_run_batch_request_multi_validation(
+    mock_emit,
+    titanic_data_context_with_fluent_pandas_and_spark_datasources_with_checkpoints_v1_with_empty_store_stats_enabled,
+    common_action_list,
+    batch_request_as_dict,
+):
+    context: FileDataContext = titanic_data_context_with_fluent_pandas_and_spark_datasources_with_checkpoints_v1_with_empty_store_stats_enabled
+
+    batch_request_0: dict = copy.deepcopy(batch_request_as_dict)
+
+    batch_request_1: dict = {
+        "datasource_name": "my_pandas_dataframes_datasource",
+        "data_asset_name": "my_dataframe_asset",
+    }
+
+    batch_request_2: dict = {
+        "datasource_name": "my_spark_filesystem_datasource",
+        "data_asset_name": "users",
+    }
+
+    batch_request_3: dict = {
+        "datasource_name": "my_spark_dataframes_datasource",
+        "data_asset_name": "my_dataframe_asset",
+    }
+
+    checkpoint: Checkpoint = Checkpoint(
+        name="my_checkpoint",
+        data_context=context,
+        config_version=1,
+        run_name_template="%Y-%M-foo-bar-template",
+        expectation_suite_name="my_expectation_suite",
+        action_list=common_action_list,
+    )
+    with pytest.raises(
+        gx_exceptions.DataContextError, match=r"expectation_suite .* not found"
+    ):
+        # noinspection PyUnusedLocal
+        result = checkpoint.run(
+            validations=[
+                {"batch_request": batch_request_0},
+                {"batch_request": batch_request_1},
+                {"batch_request": batch_request_2},
+                {"batch_request": batch_request_3},
+            ]
+        )
+
+    assert mock_emit.call_count == 1
+    # noinspection PyUnresolvedReferences
+    expected_events: List[unittest.mock._Call] = [
+        mock.call(
+            {
+                "event_payload": {
```

Comment:
This is very brittle; please parameterize this to avoid having to describe these large payloads (ie: we care that the method is called, not what the parameters are).

Reply:
@donaldheppner I put the two large `mock.call()` lists into fixtures in order to make the reading of `test_*()` methods easier.  Thanks.